### PR TITLE
feat: Migrated module `Microsoft.VirtualMachineImages/imageTemplates`

### PR DESCRIPTION
**Pull request by AlexanderSehr**
## Description

- Added module `Microsoft.VirtualMachineImages/imageTemplates`
- [x] Create AVM issue
- [x] Create GitHub teams

| Pipelines |
| - |
[avm.res.api-management.service](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.api-management.service.yml)
[avm.res.app.managed-environment](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.app.managed-environment.yml)
[avm.res.automation.automation-account](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.automation.automation-account.yml)
[avm.res.batch.batch-account](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.batch.batch-account.yml)
[avm.res.cache.redis](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.cache.redis.yml)
[avm.res.cognitive-services.account](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.cognitive-services.account.yml)
[avm.res.compute.disk-encryption-set](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.compute.disk-encryption-set.yml)
[avm.res.compute.disk](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.compute.disk.yml)
[avm.res.compute.gallery](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.compute.gallery.yml)
[avm.res.compute.proximity-placement-group](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.compute.proximity-placement-group.yml)
[avm.res.compute.ssh-public-key](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.compute.ssh-public-key.yml)
[avm.res.compute.virtual-machine](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.compute.virtual-machine.yml)
[avm.res.consumption.budget](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.consumption.budget.yml)
[avm.res.container-registry.registry](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.container-registry.registry.yml)
[avm.res.container-service.managed-cluster](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.container-service.managed-cluster.yml)
[avm.res.data-factory.factory](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.data-factory.factory.yml)
[avm.res.data-protection.backup-vault](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.data-protection.backup-vault.yml)
[avm.res.databricks.access-connector](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.databricks.access-connector.yml)
[avm.res.databricks.workspace](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.databricks.workspace.yml)
[avm.res.db-for-my-sql.flexible-server](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.db-for-my-sql.flexible-server.yml)
[avm.res.db-for-postgre-sql.flexible-server](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.db-for-postgre-sql.flexible-server.yml)
[avm.res.document-db.database-account](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.document-db.database-account.yml)
[avm.res.event-grid.domain](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.event-grid.domain.yml)
[avm.res.event-grid.system-topic](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.event-grid.system-topic.yml)
[avm.res.event-grid.topic](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.event-grid.topic.yml)
[avm.res.health-bot.health-bot](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.health-bot.health-bot.yml)
[avm.res.insights.action-group](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.insights.action-group.yml)
[avm.res.insights.activity-log-alert](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.insights.activity-log-alert.yml)
[avm.res.insights.component](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.insights.component.yml)
[avm.res.insights.data-collection-endpoint](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.insights.data-collection-endpoint.yml)
[avm.res.insights.data-collection-rule](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.insights.data-collection-rule.yml)
[avm.res.insights.diagnostic-setting](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.insights.diagnostic-setting.yml)
[avm.res.insights.metric-alert](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.insights.metric-alert.yml)
[avm.res.insights.scheduled-query-rule](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.insights.scheduled-query-rule.yml)
[avm.res.insights.webtest](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.insights.webtest.yml)
[avm.res.key-vault.vault](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.key-vault.vault.yml)
[avm.res.kubernetes-configuration.extension](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.kubernetes-configuration.extension.yml)
[avm.res.kubernetes-configuration.flux-configuration](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.kubernetes-configuration.flux-configuration.yml)
[avm.res.logic.workflow](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.logic.workflow.yml)
[avm.res.maintenance.maintenance-configuration](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.maintenance.maintenance-configuration.yml)
[avm.res.managed-identity.user-assigned-identity](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.managed-identity.user-assigned-identity.yml)
[avm.res.net-app.net-app-account](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.net-app.net-app-account.yml)
[avm.res.network.application-security-group](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.application-security-group.yml)
[avm.res.network.bastion-host](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.bastion-host.yml)
[avm.res.network.connection](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.connection.yml)
[avm.res.network.ddos-protection-plan](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.ddos-protection-plan.yml)
[avm.res.network.dns-forwarding-ruleset](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.dns-forwarding-ruleset.yml)
[avm.res.network.dns-resolver](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.dns-resolver.yml)
[avm.res.network.dns-zone](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.dns-zone.yml)
[avm.res.network.express-route-circuit](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.express-route-circuit.yml)
[avm.res.network.express-route-gateway](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.express-route-gateway.yml)
[avm.res.network.firewall-policy](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.firewall-policy.yml)
[avm.res.network.front-door-web-application-firewall-policy](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.front-door-web-application-firewall-policy.yml)
[avm.res.network.front-door](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.front-door.yml)
[avm.res.network.load-balancer](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.load-balancer.yml)
[avm.res.network.local-network-gateway](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.local-network-gateway.yml)
[avm.res.network.nat-gateways](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.nat-gateway.yml)
[avm.res.network.network-interface](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.network-interface.yml)
[avm.res.network.network-security-group](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.network-security-group.yml)
[avm.res.network.private-dns-zone](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.private-dns-zone.yml)
[avm.res.network.private-endpoint](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.private-endpoint.yml)
[avm.res.network.public-ip-address](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.public-ip-address.yml)
[avm.res.network.public-ip-prefix](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.public-ip-prefix.yml)
[avm.res.network.route-table](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.route-table.yml)
[avm.res.network.trafficmanagerprofile](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.trafficmanagerprofile.yml)
[avm.res.network.virtual-network-gateway](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.virtual-network-gateway.yml)
[avm.res.network.virtual-network](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.virtual-network.yml)
[avm.res.network.vpn-gateway](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.vpn-gateway.yml)
[avm.res.network.vpn-site](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.network.vpn-site.yml)
[avm.res.operational-insights.workspace](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.operational-insights.workspace.yml)
[avm.res.operations-management.solution](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.operations-management.solution.yml)
[avm.res.power-bi-dedicated.capacity](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.power-bi-dedicated.capacity.yml)
[avm.res.resource-graph.query](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.resource-graph.query.yml)
[avm.res.resources.deployment-script](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.resources.deployment-script.yml)
[avm.res.resources.resource-group](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.resources.resource-group.yml)
[avm.res.search.search-service](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.search.search-service.yml)
[avm.res.service-bus.namespace](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.service-bus.namespace.yml)
[avm.res.sql.server](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.sql.server.yml)
[avm.res.storage.storage-account](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.storage.storage-account.yml)
[avm.res.virtual-machine-images.image-template](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.virtual-machine-images.image-template.yml)
[avm.res.web.serverfarm](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.web.serverfarm.yml)
[avm.res.web.site](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.web.site.yml)
[avm.res.web.static-site](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.web.static-site.yml)

### CODE REVIEW

File: .github/workflows/avm.res.virtual-machine-images.image-template.yml
Comment (eriqua) [inline .github/workflows/avm.res.virtual-machine-images.image-template.yml:47]:
```suggestion
    runs-on: ubuntu-latest
```

File: avm/res/virtual-machine-images/image-template/version.json
Comment (eriqua) [inline avm/res/virtual-machine-images/image-template/version.json:3]:
```suggestion
    "version": "0.1",
```

File: avm/res/virtual-machine-images/image-template/tests/e2e/max/src/Initialize-LinuxSoftware.ps1
Comment (eriqua) [inline avm/res/virtual-machine-images/image-template/tests/e2e/max/src/Initialize-LinuxSoftware.ps1:1]: Is this script intended to stay here or to be moved to the scripts folder?

File: avm/res/virtual-machine-images/image-template/tests/e2e/max/src/Install-LinuxPowerShell.sh
Comment (eriqua) [inline avm/res/virtual-machine-images/image-template/tests/e2e/max/src/Install-LinuxPowerShell.sh:1]: Same here, we should probably move this

File: avm/utilities/pipelines/e2eValidation/resourceRemoval/Initialize-DeploymentRemoval.ps1
Comment (eriqua) [inline avm/utilities/pipelines/e2eValidation/resourceRemoval/Initialize-DeploymentRemoval.ps1:80]: Ideally, since this is impacting the removal order of all modules, we should trigger validation pipelines for all merged ones :smirk:

File: avm/res/virtual-machine-images/image-template/tests/e2e/max/src/Initialize-LinuxSoftware.ps1
Comment (AlexanderSehr) [reply avm/res/virtual-machine-images/image-template/tests/e2e/max/src/Initialize-LinuxSoftware.ps1:1]: It's intended to be here as it's a test-specific dummy script

File: avm/res/virtual-machine-images/image-template/tests/e2e/max/src/Install-LinuxPowerShell.sh
Comment (AlexanderSehr) [reply avm/res/virtual-machine-images/image-template/tests/e2e/max/src/Install-LinuxPowerShell.sh:1]: It's intended to be here as it's a test-specific dummy script

File: avm/utilities/pipelines/e2eValidation/resourceRemoval/Initialize-DeploymentRemoval.ps1
Comment (AlexanderSehr) [reply avm/utilities/pipelines/e2eValidation/resourceRemoval/Initialize-DeploymentRemoval.ps1:80]: In general I don't object, but given what this script is actually doing (function-wise) that seems like an overkill. The behavior before was, that it would (try) to remove the imageTemplate randomly before or after the MSI (which is usually removed via the resource group removal). Now that specific resource is removed before - which means no other resource is impacted. So I guess what I'm trying to say is that I would appreciate running tests for all modules only if the behavior for other modules is actually affected (like if I'd change the order if the removal - which would absolutely justify this).

Comment (eriqua) [reply avm/utilities/pipelines/e2eValidation/resourceRemoval/Initialize-DeploymentRemoval.ps1:80]: We're not modifying the behavior of this module only, but also other modules. I understand your angle, but I'm still for the suggestion. As a general rule, whenever the deployment of other modules is triggered, I'll ask for it to be done before the merge.

Comment (AlexanderSehr) [reply avm/utilities/pipelines/e2eValidation/resourceRemoval/Initialize-DeploymentRemoval.ps1:80]: Can't say I agree to generalize this, but I'll see to get it done and attach the badges. Will keep you posted 💪

File: avm/res/virtual-machine-images/image-template/tests/e2e/max/src/Initialize-LinuxSoftware.ps1
Comment (eriqua) [reply avm/res/virtual-machine-images/image-template/tests/e2e/max/src/Initialize-LinuxSoftware.ps1:1]: My understanding has always been to put scripts into the avm template folder in any case. Here I hear instead that the location of scripts depends on if its usage is strictly related to the module or not. 
Wouldn't it be easier to always put src scripts in the same folder instead of questioning each time where they should be placed? 

As minor as it may seem, if we go with those guidelines above, I'd anyway suggest to make it clear in specs or contribution guidelines how this should be done, to avoid different owners to come up with different decisions such as having an src, or source, or scripts folder placed in the test folder or anywhere else in the module.

Comment (AlexanderSehr) [reply avm/res/virtual-machine-images/image-template/tests/e2e/max/src/Initialize-LinuxSoftware.ps1:1]: Hola @eriqua,
I get where you're comming from, but I seem to have done a poor job explaining myself :D

My point is that these scripts are dummy scripts as opposed to something that are invoked by a deployment script and provide tangible value on their own - in fact they are not even used in the template but simply uploaded to a storage account to be used by a specific test case. Imagine they'd be a text file something else - we'd also not want to put them in the `assets` folder, would we? It just happens that the test cases uses a PowerShell- & Bash-based Customization-Step, as opposed to e.g. only running a Windows Restart to be a bit more 'realistic'. In that sense, they're not much more than a random artifact, as opposed to something that actually does something to the test environment.
In my mind, the point of the `assets` folder was to store IP there that you may want to re-use for other modules, say to upload scripts to a storage account, to generate SSH keys, etc. 
If we add dummy scripts, dummy text files, dummy jsons etc. there as well, we're essentially just polluting a folder for shared scripts. 
I'm also not sure why we'd need to generalize this (as you mention that we'd need to question where a script should be placed). I would struggle to come up with an example where this would be a brainteaser :D

As we find ourselves on a spectrum again, let's maybe bring it up with e.g., @ChrisSidebotham and resolve it one way or another. 💪

Comment (ChrisSidebotham) [reply avm/res/virtual-machine-images/image-template/tests/e2e/max/src/Initialize-LinuxSoftware.ps1:1]: My view and practice of this with TF in Customer env's I would be happy where the script currently is. if the script is contained to that singular test file and cannot be used by other resources then it should reside within the folder structure for the test case. Also keeping it clear to the end user when they review of test files for usage examples (which does happen) they can see what "assets" have been used to achieve the configuration

File: avm/utilities/pipelines/e2eValidation/resourceRemoval/Initialize-DeploymentRemoval.ps1
Comment (AlexanderSehr) [reply avm/utilities/pipelines/e2eValidation/resourceRemoval/Initialize-DeploymentRemoval.ps1:80]: @eriqua you may not have noticed, but I did that back then ;)

File: avm/res/virtual-machine-images/image-template/tests/e2e/max/src/Initialize-LinuxSoftware.ps1
Comment (AlexanderSehr) [reply avm/res/virtual-machine-images/image-template/tests/e2e/max/src/Initialize-LinuxSoftware.ps1:1]: Hey @eriqua please let me know what your thoughts are / if they changed in any way.

Comment (eriqua) [reply avm/res/virtual-machine-images/image-template/tests/e2e/max/src/Initialize-LinuxSoftware.ps1:1]: If you both agree I'll step back. I'm overall fine with the decision. I'd only have appreciated to have it written down as straightforward as it may sound. I'll just create an issue in AVM and unblock this PR.

**Review by ChrisSidebotham — DISMISSED** [review]
👌

File: .github/workflows/avm.res.virtual-machine-images.image-template.yml
Comment (eriqua) [inline .github/workflows/avm.res.virtual-machine-images.image-template.yml:75]: If the gh token fix gets approved, my last comment :)
```suggestion
    name: "Run"
    permissions:
      id-token: write # For OIDC
      contents: write # For release tags
    needs:
```

Comment (AlexanderSehr) [reply .github/workflows/avm.res.virtual-machine-images.image-template.yml]: updated

**Review by eriqua — DISMISSED** [review]
🆗

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**
